docs/design/segments.md

```diff
@@ -209,7 +207,7 @@ For example, if your real-time ingestion creates 3 segments that were sharded wi
 
 ## Schema changes
 
-## Replacing segments
+### Replacing segments
```

Comment:
@vtlim , @jill-imply This is a little hairier than a simple heading adjustment I'm afraid.

"Schema Changes" doesn't really convey the content here. There seem to be two scenarios:
- different segment versions that have different schemas
- differing schemas within a single segment version

Couple other thoughts:
- Normally we shouldn't have two headings back to back.
- think we should improve the "Replacing segments" heading because there is a lot of explaining of the segment naming convention before we get to the fact that the new version has a different schema than the old version

Reply:
I wonder if we could combine this material with the "Naming Convention" section and name that "Segment identification". The "Replacing segments" section repeats some of the same content.

Reply:
Updated - ready for review.